src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,146 @@
+                BigDecimal price = resultSet.getObject(THIRD_COLUMN_INDEX, BigDecimal.class);
+
+                Book book = new Book(identify, title, price);
+                optionalBook = Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to establish connection"
+                    + " with the database", e);
+        }
+        return optionalBook;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> bookList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(SQL_OUTPUT_REQUEST)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            while (resultSet.next()) {
+                Long id = resultSet.getObject(FIRST_COLUMN_INDEX, Long.class);
+                String title = resultSet.getObject(SECOND_COLUMN_INDEX, String.class);
+                BigDecimal price = resultSet.getObject(THIRD_COLUMN_INDEX, BigDecimal.class);
+
+                Book book = new Book(id, title, price);
+                bookList.add(book);
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to establish connection"
+                    + " with the database", e);
+        }
+        return bookList;
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(SQL_UPDATE_REQUEST)) {
+            preparedStatement.setString(FIRST_PARAMETER_INDEX, book.getTitle());
+            preparedStatement.setBigDecimal(SECOND_PARAMETER_INDEX, book.getPrice());
+            preparedStatement.setLong(THIRD_PARAMETER_INDEX, book.getId());
+
+            int affectedRows = preparedStatement.executeUpdate();
+
+            if (affectedRows < 1) {
+                throw new DataProcessingException("No book found with ID: " + book.getId());
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to establish connection"
+                    + " with the database", e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        int affectedRows = 0;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(SQL_DELETE_REQUEST)) {
+            preparedStatement.setLong(FIRST_PARAMETER_INDEX, id);
+            affectedRows = preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to establish "
+                    + "connection with the database", e);
+        }
+        return affectedRows > 0;
```

Comment:
Don't return `true` all the time in the `deleteById` method. Instead, return the boolean value depending on the `preparedStatement.executeUpdate()` result. The current implementation is correct, but ensure that this logic is maintained if changes are made.